admin/abstract/views.py

```diff
@@ -0,0 +1,49 @@
+from django.views.generic import FormView
+from django.http import HttpResponseNotFound
+from django.shortcuts import render
+
+
+class GuidFormView(FormView):
+    form_class = None
+    template_name = None
+    object_type = None
+
+    def get_guid_object(self):
+        raise NotImplementedError
+
+    def __init__(self):
+        self.guid = None
+        super(GuidFormView, self).__init__()
+
+    def get(self, request, *args, **kwargs):
+        self.guid = request.GET.get('guid', None)
+        if self.guid is not None:
+            try:
+                guid_object = self.get_guid_object()
+            except (AttributeError, TypeError):
+                return HttpResponseNotFound(
```

Comment:
Ah, okay. By default, implementing `page_not_found` will just send back a standard 404 with a default message. https://docs.djangoproject.com/en/1.9/ref/views/#django.views.defaults.page_not_found

If you create a 404.html in the root of your templates directory, it will be loaded and you should have the exception raised in the template context. 

Alternatively, you can set handler404 to any FBV that you like and it will be called when a 404 exception is raised. I believe it will be passed 404.html as it's template, just like page_not_found is.

The easiest (and probably most sustainable) thing to do would be to make a 404.html that extends the base layout like the rest of the templates do and displays the exception message and then switch back to `page_not_found`.